### PR TITLE
Added a demand checker for Azure Pipelines

### PR DESCRIPTION
**Pull request by Eldarrin**
Added a parameter to the metadata so that you can specify a parent template which must be the ADO agent name of the offline agent.

Job now compares the template name to determine if it is matched to a potential agent

Issues: If more than one agent can fulfil the job, multiple jobs can be instantiated as they decide independently that they are able to fulfil.

Docs: https://github.com/kedacore/keda-docs/pull/685

Andy

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [ ] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [ ] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Relates to https://github.com/kedacore/keda/issues/2328

### CODE REVIEW

File: pkg/scalers/azure_pipelines_scaler.go [hunk 1]
```diff
@@ -41,6 +41,8 @@ type azurePipelinesMetadata struct {
 	organizationURL            string
```
Comment (tomkerkhove) [inline pkg/scalers/azure_pipelines_scaler.go:41]: Can you update the changelog please?

File: pkg/scalers/azure_pipelines_scaler.go [hunk 2]
```diff
@@ -119,6 +127,7 @@ func parseAzurePipelinesMetadata(ctx context.Context, config *ScalerConfig, http
 	}
 
 	meta.scalerIndex = config.ScalerIndex
+	meta.scalarName = config.Name
```
Comment (zroubalik) [inline pkg/scalers/azure_pipelines_scaler.go:130]: What is the purpose of this? Also there's a typo `scalar` -> `scaler`

File: CHANGELOG.md [hunk 3]
```diff
@@ -37,6 +37,7 @@
 - **Azure Queue:** Don't call Azure queue GetProperties API unnecessarily ([#2613](https://github.com/kedacore/keda/pull/2613))
 - **Datadog Scaler:** Validate query to contain `{` to prevent panic on invalid query ([#2625](https://github.com/kedacore/keda/issues/2625))
 - **Kafka Scaler** Make "disable" a valid value for tls auth parameter ([#2608](https://github.com/kedacore/keda/issues/2608))
+- **Azure Pipelines:** Added the ability to support ADO demands by referencing a master template ([#2707](https://github.com/kedacore/keda/pull/2707))
```
Comment (zroubalik) [inline CHANGELOG.md:40]: Please sort alphabetically

**Review by zroubalik — COMMENTED** [review]
Would you mind addind some tests?

File: CHANGELOG.md (hunk 3 above)
Comment (Eldarrin) [reply CHANGELOG.md:40]: Just returned from vacation so shall look into that this week

**Comment by zroubalik** [comment]
@Eldarrin please check you PR, seems like there is something broken and there are included changes from another PRs.

**Comment by Eldarrin** [comment]
Seems like my fork is corrupted with a previous merge. May be best if I kill the fork, You close the PR. I'll open a new PR with a clean fork and reference this one.

**Pull request closed without merging**